homeassistant/components/smartthings/light.py

```diff
@@ -0,0 +1,215 @@
+            features |= \
+                SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+        # Color Temperature
+        if Capability.color_temperature in self._device.capabilities:
+            features |= SUPPORT_COLOR_TEMP
+        # Color
+        if Capability.color_control in self._device.capabilities:
+            features |= SUPPORT_COLOR
+
+        return features
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn the light on."""
+        tasks = []
+        # Color temperature
+        if self._supported_features & SUPPORT_COLOR_TEMP \
+           and ATTR_COLOR_TEMP in kwargs:
+            tasks.append(self.set_color_temp(
+                kwargs[ATTR_COLOR_TEMP]))
+        # Color
+        if self._supported_features & SUPPORT_COLOR \
+           and ATTR_HS_COLOR in kwargs:
+            tasks.append(self.set_color(
+                kwargs[ATTR_HS_COLOR]))
+        if tasks:
+            # Set temp/color first
+            await asyncio.gather(*tasks)
+
+        # Switch/brightness/transition
+        if self._supported_features & SUPPORT_BRIGHTNESS \
+           and ATTR_BRIGHTNESS in kwargs:
+            await self.set_level(
+                kwargs[ATTR_BRIGHTNESS],
+                kwargs.get(ATTR_TRANSITION, 0))
+        else:
+            await self._device.switch_on(set_status=True)
+
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn the light off."""
+        # Switch/transition
+        if self._supported_features & SUPPORT_TRANSITION \
+                and ATTR_TRANSITION in kwargs:
+            await self.set_level(0, int(kwargs[ATTR_TRANSITION]))
+        else:
+            await self._device.switch_off(set_status=True)
+
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_update(self):
+        """Update entity attributes when the device status has changed."""
+        # Brightness and transition
+        if self._supported_features & SUPPORT_BRIGHTNESS:
+            self._brightness = convert_scale(
+                self._device.status.level, 100, 255)
+        # Color Temperature
+        if self._supported_features & SUPPORT_COLOR_TEMP:
+            self._color_temp = color_util.color_temperature_kelvin_to_mired(
+                self._device.status.color_temperature)
+        # Color
+        if self._supported_features & SUPPORT_COLOR:
+            self._hs_color = (
+                convert_scale(self._device.status.hue, 100, 360),
+                self._device.status.saturation
+            )
+
+    async def set_color(self, hs_color):
```

Comment:
We're using the convention to prefix async and callback method names with `async_`